### ISSUE TITLE
VM: Unexpected deep copy of ref variables

### ISSUE THREAD

**Issue by flyx** [nim-lang/Nim#1330]
``` nimrod
static:
    var a: ref string
    new(a)
    a[] = "test1"
    var b = a
    b[] = "test2"
    echo a[]
    echo b[]
```

Expected output at compile time:

```
test2
test2
```

Actual output:

```
test1
test2
```

There seems to be a deep copy happening; `a` and `b` reference different strings.

**Comment by jangko** [comment]
this already fixed, no longer a valid issue for nim ver 0.12.0

**Comment by reactormonk** [comment]
Just tested, still the incorrect result.

**Comment by jangko** [comment]
oops, sorry

**Comment by Araq** [comment]
Still broken.

**Comment by GULPF** [comment]
Fixed by #7797
